core/server/common/src/main/java/alluxio/master/journal/AsyncJournalWriter.java

```diff
@@ -308,6 +312,10 @@ private void doFlush() {
           }
         }
       } catch (IOException | JournalClosedException exc) {
+        // Add the error logging here since the actual flush error may be overwritten
+        // by the future meaningless ratis.protocol.AlreadyClosedException
+        // use debug level logging because the exception may be annoying in some cases
+        LOG.debug("Failed to flush journal entry", exc);
```

Comment:
(nit) consider using `SamplingLogger` if the error happens too frequently.

Reply:
Good advice! I don't know we have that convenient logging way!